src/index.ts

```diff
@@ -11,7 +12,14 @@ import { UpdateTimeReleaser } from './updateTime/updateTimeReleaser';
 import { HeartbeatReleaser } from './heartbeat/heartbeatReleaser';
 import { ExpirationStatusUpdater } from './taskExpiration/expirartionStatusUpdater';
 
+let isActive = false;
+
 async function run(logger: Logger): Promise<void> {
+  if (isActive) {
```

Comment:
why do you need the "isActive" flag ?
what is the need of this?

Reply:
Because in case the interval time is arrived and the service is still in-progress, the "run" function will know not to be executed

Reply:
So why don't you use timers/promises in order to make sure the interval finishes before calling a new one?
https://nodejs.org/docs/v22.14.0/api/timers.html#timerspromisessetintervaldelay-value-options